src/age/stream-cipher.ts

```diff
@@ -12,6 +12,10 @@ const TAG_SIZE = 16 // Poly1305 MAC size
 const ENCRYPTED_CHUNK_SIZE = CHUNK_SIZE + TAG_SIZE
 const NONCE_SIZE = 12 // STREAM nonce size
 
+// due to using a 32bit uint for the counter, this is the max
+// value the counter can be without risking a nonce reuse
+const COUNTER_MAX = Math.pow(2, 32)
```

Comment:
The Max uint32 value is 4294967295, after which is would wrap.
Are you sure you want 4294967296 here ? (which is `0b100000000000000000000000000000000`, thus bigger than what a uint32 can hold.)

Reply:
You make a very valid point!